guidance/library/_capture.py

```diff
@@ -1,6 +1,9 @@
 from .._guidance import guidance
 from .._grammar import capture as grammar_capture, GrammarFunction
 
+# Adapted from active_role_end in _model.py, functionality should be shared probably?
+import re
+format_pattern = re.compile(r"<\|\|_.*?_\|\|>", flags=re.DOTALL)
```

Comment:
Slightly concerned that this appears to be relying on ChatML tags, which not all models use